src/api/spotify-api.ts

```diff
@@ -0,0 +1,54 @@
+import { refreshAccessToken } from '../main/auth';
+
+const API_URL = 'https://api.spotify.com/v1';
+
+class SpotifyApi {
+  private isThrottled: boolean;
+  private throttleTime: number;
+
+  public refreshToken: string;
+
+  async fetch(input: RequestInfo, init?: RequestInit) {
+    if (this.isThrottled) {
+      const timeLeft = this.throttleTime - new Date().getTime();
+      if (timeLeft > 0) {
+        console.warn(
+          `Spotify API call ignored, waiting ${timeLeft}ms... (${input})`
```

Comment:
I haven't tested yet, but I wonder how this will work with multiple requests going on concurrently. I guess one might win the race, and then the other will have to keep on waiting? Technically, I think one would want to also pass some request `priority` and ensure that the higher priority requests take precedence over lower priority requests -- for example, changing volume might be lower priority than the heartbeat.

For a toy like Lofi, this is overkill however 😄

Reply:
Yeah, we could have a queue handling all of this but it is overkill. In order to get a 429, just mess with the scroll wheel (volume change) like there's no tomorrow, you'll get a 429 in a few seconds.